bgpd/bgp_io.c

```diff
@@ -151,7 +151,7 @@ static int bgp_process_writes(struct thread *thread)
 	if (reschedule) {
 		thread_add_write(fpt->master, bgp_process_writes, peer,
 				 peer->fd, &peer->t_write);
-	} else if (!fatal) {
+	} else if (!fatal && peer->v_routeadv == 0) {
```

Comment:
This doesn't belong here, put the check in `bgp_generate_updgrp_packets` please.

Reply:
bgp_generate_updgrp_packet is also invoked by MRAI func bgp_adjust_routeadv. And bgp_adjust_routeadv will kick off the  bgp_generate_updgrp_packet thread if the v_routeadv is 0, so IMO, it is hard to check whether  bgp_generate_updgrp_packet is stared by MRAI or not without adding a new parameter. 

BTW, I'm not familiar with the background,  IMO, MARI handles  the bgp_generate_updgrp_packet thread correctly, why this bgp_generate_updgrp_packet is also invoked by the bgp IO ?  According to my understanding, there is no need to kick off bgp_generate_updgrp_packet thread here.

Reply:
We can be at function bgp_generate_updgrp_packets from many other palces as well. So right palce to have this check is inside function bgp_generate_updgrp_packets. If timer is still running then we can return without sending any update.

Reply:
@Spantik Thanks for your explanation. I updated the PR, change the check into function bgp_generate_updgrp_packets.